app/app.go

```diff
@@ -127,6 +127,33 @@ func NewApp(c appConfig, db *sql.DB) (*App, error) {
 		}
 	}
 
+	if c.TLSListenAddr != "" {
+		var cert tls.Certificate
+		if c.TLSCertFile != "" && c.TLSKeyFile != "" {
+			cert, err = tls.LoadX509KeyPair(c.TLSCertFile, c.TLSKeyFile)
+			if err != nil {
+				return nil, errors.Wrap(err, "load tls cert file")
+			}
+		}
+		//if both TLSCertFile and TLSCert are set in config, TLSCert will be used
+		if c.TLSCert != "" && c.TLSKey != "" {
+			cert, err = tls.X509KeyPair([]byte(c.TLSCert), []byte(c.TLSKey))
+			if err != nil {
+				return nil, errors.Wrap(err, "parse tls cert")
+			}
+		}
+
+		cfg := &tls.Config{Certificates: []tls.Certificate{cert}}
+
+		l2, err := tls.Listen("tcp", c.TLSListenAddr, cfg)
+		if err != nil {
+			return nil, errors.Wrapf(err, "bind address %s", c.TLSListenAddr)
+		}
+		app.l = newMultiListener(l, l2)
+	} else {
+		app.l = newMultiListener(l)
```

Comment:
Changing the app to just use `net.Listener` for it's struct field eliminates the need for this `else`